src/main/java/org/apache/sling/pipes/internal/MovePipe.java

```diff
@@ -63,23 +67,38 @@ public boolean modifiesContent() {
+                        Resource target = resolver.getResource(targetPath);
+                        Resource parent = target.getParent();
                         Node targetParent = session.getItem(targetPath).getParent();
-                        if (targetParent.getPrimaryNodeType().hasOrderableChildNodes()) {
-                            String targetPathNewNode = targetPath + UUID.randomUUID() ;
-                            session.move(resource.getPath(), targetPathNewNode);
-                            String newNodeName = targetPathNewNode.substring(targetPathNewNode.lastIndexOf("/") + 1);
-                            String oldNodeName = targetPath.substring(targetPath.lastIndexOf("/") + 1);
-                            targetParent.orderBefore(newNodeName, oldNodeName);
-                            session.removeItem(targetPath);
-                            // Need to use JackrabbitNode.rename() here, since session.move(targetPathNewNode, targetPath)
-                            // would move the new node back to the end of its siblings list
-                            JackrabbitNode newNode = (JackrabbitNode) session.getNode(targetPathNewNode);
-                            newNode.rename(oldNodeName);
+                        String targetPathNewNode = targetPath + UUID.randomUUID();
+                        String newNodeName = targetPathNewNode.substring(targetPathNewNode.lastIndexOf("/") + 1);
+                        String oldNodeName = targetPath.substring(targetPath.lastIndexOf("/") + 1);
+                        if (orderBefore) {
+                            logger.debug("ordering {} before {}", resource.getPath(), targetPath);
+                            if (targetParent.getPrimaryNodeType().hasOrderableChildNodes()) {
+                                String targetNodeName = ResourceUtil.createUniqueChildName(parent, resource.getName());
+                                String targetNodePath = targetParent.getPath() + SLASH + targetNodeName;
+                                session.move(resource.getPath(), targetNodePath);
+                                targetParent.orderBefore(targetNodeName, oldNodeName);
+                                output = Collections.singleton(parent.getChild(targetNodeName)).iterator();
+                            } else {
+                                logger.warn("parent resource {} doesn't support ordering", target.getPath());
+                            }
                         } else {
-                            session.removeItem(targetPath);
-                            session.move(resource.getPath(), targetPath);
+                            logger.debug("overwriting {}", targetPath);
+                            if (targetParent.getPrimaryNodeType().hasOrderableChildNodes()) {
+                                session.move(resource.getPath(), targetPathNewNode);
+                                targetParent.orderBefore(newNodeName, oldNodeName);
```

Comment:
why orderBefore? aren't we in the case where there is no "orderBefore" configured?

Reply:
@npeltier  , this part of code was already there which gets triggered when we have overwrite flag set to true , so this basically overwrites the target node .